.github/workflows/conda-envs/test-oldest_env.yaml

```diff
@@ -0,0 +1,12 @@
+name: test
+channels:
+  - conda-forge
+  - defaults
+
+dependencies:
+  - tbb
```

Comment:
`tbb`, `tbb-devel` should be pinned to the minimum supported version here. [The docs claim](https://freud.readthedocs.io/en/latest/gettingstarted/installation.html) compatibility with `>=2017.03.R2` but I am unsure if that is still accurate since the TBB 2020 updates.

(I don't really care about pinning ninja/pip since they're not build dependencies. One could make the argument that pip matters because various behaviors have changed over time, but I don't feel strongly about pinning it.)

Reply:
Looking through tbb's versions on conda-forge, I don't see a version matching that name. In fact, the development versions only go back as far as `2018.0.4`. I'll try to pin to that version for now and see what happens. We'll have to update the oldest supported version in our docs.